pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsDataDriven failed

pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test.TestSQLStatsDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20370645?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20370645?buildTab=artifacts#/) on master @ [19606f2d7f4edb6ff39eba4df1773ab6fc68d1dc](https://github.com/cockroachdb/cockroach/commits/19606f2d7f4edb6ff39eba4df1773ab6fc68d1dc):


```
          SELECT 1, 1
          SELECT 1, 1
          COMMIT
          ----
    datadriven_test.go:190: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16693/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_/persistedsqlstats_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/sqlstats/persistedsqlstats/testdata/stmt_grouping_in_explicit_txn:21:
          exec-sql
          BEGIN
          SELECT 1
          SELECT 1, 1
          SELECT 1, 1
          SELECT 1, 1
          COMMIT
          ----
    datadriven_test.go:190: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16693/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_/persistedsqlstats_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/sqlstats/persistedsqlstats/testdata/stmt_grouping_in_explicit_txn:30:
          exec-sql
          BEGIN
          SELECT 1
          SELECT 1, 1
          SELECT 1, 1
          SELECT 1, 1
          COMMIT
          ----
    datadriven_test.go:190: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16693/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_/persistedsqlstats_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/sqlstats/persistedsqlstats/testdata/stmt_grouping_in_explicit_txn:39:
          wait-for-txn-stats app=test count=3 execCount=4
          ----
    datadriven_test.go:190: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16693/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_/persistedsqlstats_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/sqlstats/persistedsqlstats/testdata/stmt_grouping_in_explicit_txn:45:
          observe-sql
          SELECT
            encode(fingerprint_id, 'hex'),
            metadata -> 'stmtFingerprintIDs'
          FROM
            crdb_internal.transaction_statistics
          WHERE
            app_name = 'test' AND
            jsonb_array_length(metadata -> 'stmtFingerprintIDs') >= 3
          ORDER BY
            jsonb_array_length(metadata -> 'stmtFingerprintIDs')
          ASC
        
        expected:
          dce4fa24f23a3c31,["1d2a16425a50b888", "647cd604b055a7b7", "647cd604b055a7b7"]
          f72debc328a1e0f4,["1d2a16425a50b888", "647cd604b055a7b7", "647cd604b055a7b7", "647cd604b055a7b7"]
        
        found:
          dce4fa24f23a3c31,["1d2a16425a50b888", "647cd604b055a7b7", "647cd604b055a7b7"]
--- FAIL: TestSQLStatsDataDriven/stmt_grouping_in_explicit_txn (0.95s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/sqlstats/persistedsqlstats/datadriven_test.go
+++ b/pkg/sql/sqlstats/persistedsqlstats/datadriven_test.go
@@ -76,6 +76,7 @@ func TestSQLStatsDataDriven(t *testing.T) {
 	knobs.StubTimeNow = stubTime.Now
 	knobs.OnStmtStatsFlushFinished = injector.invokePostStmtStatsFlushCallback
 	knobs.OnTxnStatsFlushFinished = injector.invokePostTxnStatsFlushCallback
+	knobs.SynchronousSQLStats = true
 	params.Knobs.SQLStatsKnobs = knobs
 
 	cluster := serverutils.StartCluster(t, 3 /* numNodes */, base.TestClusterArgs{

